web/src/message_events.js

```diff
@@ -208,6 +209,12 @@ export function update_messages(events) {
         }
 
         if (topic_edited || stream_changed) {
+            // Clear message list data cache since the local data for the
+            // filters might no longer be accurate.
+            //
+            // TODO: Add logic to update the message list data cache.
+            message_list_data_cache.clear();
```

Comment:
This temporary code might be wrong, in that I think `all_messages_data` gets permanently ejected?

Reply:
oh yeah! Nice catch.

Reply:
Using cache = [all_messages_data.all_messages_data]; now